src/providers/mod.rs

```diff
@@ -195,19 +195,38 @@ fn write_ssh_keys(user: User, ssh_keys: Vec<PublicKey>) -> Result<()> {
 }
 
 pub trait MetadataProvider {
-    fn attributes(&self) -> Result<HashMap<String, String>>;
-    fn hostname(&self) -> Result<Option<String>>;
-    fn ssh_keys(&self) -> Result<Vec<PublicKey>>;
-    fn networks(&self) -> Result<Vec<network::Interface>>;
-    fn boot_checkin(&self) -> Result<()>;
+    fn attributes(&self) -> Result<HashMap<String, String>> {
+        let attributes = maplit::hashmap! {};
+        Ok(attributes)
```

Comment:
Nit: `Ok(HashMap::new())` should be enough here (it also avoids a macro-expansion pass).

Reply:
Indeed.... my rust noob status is showing :) I had wondered about going with anonymous hashmap there. The tip is appreciated.